src/main/java/network/brightspots/rcv/TieBreak.java

```diff
@@ -312,42 +293,63 @@ private String doRandom() {
   }
 
   // function doPreviousRounds
-  // purpose: select candidate based on previous round tallies
+  // purpose: select candidate based on previous round tallies (with a fallback to interactive or
+  //   random if necessary)
+  // param: tiedCandidates is the list of candidates in contention to be selected
   // returns: selected candidate
-  private String doPreviousRounds() {
+  private String doPreviousRounds(List<String> tiedCandidates) throws TabulationCancelledException {
     String selection = null;
     // stores candidates still in contention while we iterate through previous rounds to determine
     // the selection
-    Set<String> candidatesInContention = new HashSet<>(tiedCandidates);
+    List<String> candidatesInContention = tiedCandidates;
 
-    for (int round = this.round - 1; round > 0; round--) {
+    for (int roundToCompare = this.round - 1; roundToCompare > 0; roundToCompare--) {
       // map of tally to candidate IDs for round under consideration
       SortedMap<BigDecimal, LinkedList<String>> tallyToCandidates =
-          Tabulator.buildTallyToCandidates(roundTallies.get(round), candidatesInContention, false);
+          Tabulator.buildTallyToCandidates(
+              roundTallies.get(roundToCompare), new HashSet<>(candidatesInContention), false);
       BigDecimal voteTotalForSelection =
           selectingAWinner ? tallyToCandidates.lastKey() : tallyToCandidates.firstKey();
-      LinkedList<String> candidatesWithVoteTotal = tallyToCandidates.get(voteTotalForSelection);
-      if (candidatesWithVoteTotal.size() == 1) {
-        selection = candidatesWithVoteTotal.getFirst();
+      candidatesInContention = tallyToCandidates.get(voteTotalForSelection);
+      if (candidatesInContention.size() == 1) {
+        selection = candidatesInContention.get(0);
         explanation =
             String.format(
                 "%s had the %s votes (%s) in round %d.",
                 selection,
                 selectingAWinner ? "most" : "fewest",
                 voteTotalForSelection.toString(),
-                round);
+                roundToCompare);
         break;
+      } // else keep looping
+    }
+
+    // if 2 or more candidates are still tied, we fall back to interactive or random with the
+    // remaining candidates
+    if (candidatesInContention.size() > 1) {
+      String explanationPrefix =
+          String.format(
+              "Comparing previous round counts still resulted in a tie %s %s, so we fell back to %s.",
+              candidatesInContention.size() > 2 ? "among" : "between",
+              Utils.listToSentenceWithQuotes(candidatesInContention),
+              tieBreakMode == TieBreakMode.PREVIOUS_ROUND_COUNTS_THEN_RANDOM
+                  ? "random"
+                  : "interactive");
+      if (tieBreakMode == TieBreakMode.PREVIOUS_ROUND_COUNTS_THEN_INTERACTIVE) {
+        selection = doInteractive(candidatesInContention);
```

Comment:
I hope someday, somewhere, somehow, this line of code is executed in a non-test tabulation and gets its wings.

Reply:
I'll take the under.